File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/string/StringEmptyRule.java
package com.example.navigation.stepsEngine.flow.rules.comparable.string;

import com.example.navigation.stepsEngine.flow.rules.comparable.ComparableRule;
import com.example.navigation.stepsEngine.payment.FlowState;

public class StringEmptyRule extends ComparableRule {

    @Override
    public boolean evaluate(FlowState flowState) {
        return ((String) getValue(flowState)).isEmpty();
    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/string/StringEmptyRule.kt
package com.example.navigation.stepsEngine.flow.rules.comparable.string

import com.example.navigation.stepsEngine.flow.rules.comparable.ComparableRule
import com.example.navigation.stepsEngine.payment.FlowState

class StringEmptyRule : ComparableRule() {
    override fun evaluate(flowState: FlowState): Boolean {
        return (getValue(flowState) as String?)!!.isEmpty()
    }
}